drivers/ethernet/eth_stm32_hal.c

```diff
@@ -1138,7 +1149,12 @@ static void set_mac_config(const struct device *dev, struct phy_link_state *stat
 	mac_config.DuplexMode =
 		PHY_LINK_IS_FULL_DUPLEX(state->speed) ? ETH_FULLDUPLEX_MODE : ETH_HALFDUPLEX_MODE;
 
-	mac_config.Speed = PHY_LINK_IS_SPEED_100M(state->speed) ? ETH_SPEED_100M : ETH_SPEED_10M;
+	mac_config.Speed =
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32n6_ethernet)
+		PHY_LINK_IS_SPEED_1000M(state->speed) ? ETH_SPEED_1000M :
+#endif
```

Comment:
Can `IF_ENABLED(DT_HAS_COMPAT_STATUS_OKAY(st_stm32n6_ethernet)` could also be used here ?